fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java

```diff
@@ -213,7 +213,7 @@ public class GsonUtils {
     private static final RuntimeTypeAdapterFactory<HeartbeatResponse> HEARTBEAT_RESPONSE_ADAPTER_FACTOR =
             RuntimeTypeAdapterFactory
                     .of(HeartbeatResponse.class, "clazz")
-                    .registerSubtype(BackendHbResponse.class, BackendHbResponse.class.getSimpleName())
+                    .registerSubtype(DataNodeHbResponse.class, DataNodeHbResponse.class.getSimpleName())
```

Comment:
BackendHbResponse class name is serialized in json, and rename will cause compatibility issue.